### DISCUSSION TITLE
从包管理安装 wiliwili

### DISCUSSION THREAD

**Discussion by xfangfang** [xfangfang/wiliwili#92] [Q&A]
此页面收集各系统下直接通过包管理安装 wiliwili 的方法

系统不限，欢迎大家制作提交，可以直接评论在下方

**Comment by xfangfang** [comment]
# archLinux

作者： [yuioto](https://github.com/yuioto)
相关讨论：https://github.com/xfangfang/wiliwili/discussions/71
相关链接：https://aur.archlinux.org/packages/wiliwili-git

请参考本层楼中楼内说明使用

**Reply by Puqns67** [reply]
希望能更新为以下内容：

# Arch Linux

## 通过 archlinuxcn 源进行安装

1. 添加 archlinuxcn 源至系统，添加方法与相关介绍参见下方链接：（如已添加过可跳过此步骤）
https://www.archlinuxcn.org/archlinux-cn-repo-and-mirror/

3. 使用 Pacman 安装 wiliwili
```sh
pacman -S wiliwili
```

## 通过 AUR (Arch User Repository) 安装

AUR 中的 wiliwili 分为三个版本，相关信息可点击版本名查看，分别如下（括号中为对应的包名）：

[正式版](https://aur.archlinux.org/packages/wiliwili)（wiliwili），[主线版](https://aur.archlinux.org/packages/wiliwili-git)（wiliwili-git），[预编译版](https://aur.archlinux.org/packages/wiliwili-bin)（wiliwili-bin）

* 常见 AUR 助手有 paru 与 yay，请按照实际情况在自行在下方命令中替换。
* 推荐使用正式版。由于目前无人维护，不推荐预编译版。

```sh
<AUR 助手> -S <包名>
```

例子：
```sh
# 使用 paru 安装稳定版
paru -S wiliwili

# 使用 yay 安装主线版
yay -S wiliwili-git
```

**Reply by xfangfang** [reply]
感谢更新文档，不如就保留着你的回复，这样其他人也是能直接看到的，避免我再复制一份的话浪费篇幅，感谢感谢。

**Comment by xfangfang** [comment]
# fedora

作者：[mochaaP](https://github.com/mochaaP)
相关讨论：#143
相关链接：https://copr.fedorainfracloud.org/coprs/mochaa/wiliwili/

**Comment by happytommyl** [comment]
# Scoop
Now Wiliwili is in scoop/Extras on Windows

To install on Windows:
`scoop bucket add Extras`
`scoop install wiliwili` 

https://github.com/ScoopInstaller/Extras/pull/11453

**Comment by sanweiya** [comment]
# openSUSE (OBS)
```
sudo zypper install opi

opi wiliwili
```

**Comment by xfangfang** [comment]
# Flatpak

作者：[madoka773](https://github.com/madoka773)
相关讨论：https://github.com/xfangfang/wiliwili/issues/252
相关链接：https://flathub.org/apps/cn.xfangfang.wiliwili

```
flatpak install flathub cn.xfangfang.wiliwili
```

**Comment by pugaizai** [comment]
# Voidlinux
## 如何使用

在此之前请安装 `xtools` 以方便以后的操作：

```
# xbps-install -S xtools
```

请克隆voidlinux-zh仓库

```
$ git clone https://github.com/voidlinux-zh/void-packages-zh --depth=1 
```

然后选择wiliwili进行编译：

```
$ cd void-packages-zh
$ ./xbps-src binary-bootstrap
$ ./xbps_src pkg wiliwili
```

编译完了，就可以安装了：

```
$ xi wiliwili #确保是在 void-packages-zh 目录中
```

**Comment by Aleksanaa** [comment]
# NixOS

## 如何使用

临时安装：

```
# nix shell nixpkgs#wiliwili
```

用nix profile安装：

```
# nix profile install nixpkgs#wiliwili
```

在配置中添加（推荐）：

```nix
users.users.<username>.packages = with pkgs; [
  wiliwili
];
```

或者

```nix
environment.systemPackages = with pkgs; [
  wiliwili
];
```

后续会支持在macOS上用Nix安装wiliwili，但目前暂时swift框架构建出错，因此没有适配

**Comment by happytommyl** [comment]
# winget
作者：[Dragon1573](https://github.com/Dragon1573)
相关链接：https://github.com/microsoft/winget-pkgs/pull/171591
相关讨论：#382 
`winget install wiliwili`

**Reply by Dragon1573** [reply]
@xfangfang 大佬你好👋🏼 

我注意到此应用直接借助 **GitHub Releases** 实现发行。对于此类应用，`microsoft/winget-pkgs` 社区成员专门开发了 [WinGet Releaser](https://github.com/marketplace/actions/winget-releaser) ，方便应用的官方维护团队将「同步发行至 WinGet」这个步骤集成到 CI/CD 流水线中。

- 参考模范：[Oh My Posh](https://github.com/JanDeDobbeleer/oh-my-posh/actions/workflows/winget.yml)
- 参考模范：[Komac](https://github.com/russellbanks/Komac/actions/workflows/release.yml)

大佬是否有意向在本仓库的 `.github/workflows/` 中集成它，由您以自动化形式完全接管本应用在 `microsoft/winget-pkgs` 上的后续发行呢？🙏🏼

**Reply by Dragon1573** [reply]
> 大佬是否有意向在本仓库的 `.github/workflows/` 中集成它，由您以自动化形式完全接管本应用在 `microsoft/winget-pkgs` 上的后续发行呢？🙏🏼

由社区成员自发驱动的 _WinGet_ 元信息更新不可避免存在滞后性，且由于社区志愿者成员的「身份性质」，不可避免地存在较长的人工虚拟环境审核过程。

当元信息被此软件的官方维护团队接管后，官方维护团队将在某种程度上获得「特权」，版本升级请求将被更快地合并至仓库主分支并刷新到 API 端点向全球 _WinGet_ 用户推送，这将显著降低 _WinGet_ 源与 _GitHub Releases_ 之间的滞后性。

**Reply by xfangfang** [reply]
感谢提醒和之前的winget提交贡献。
我对winget不是特别了解，感觉会浪费很多时间在熟悉这个工具上，如果您有经验的话可以将相关的pr提交过来，有我需要做的什么事随意提哈～

**Reply by Dragon1573** [reply]
> 如果您有经验的话可以将相关的pr提交过来，有我需要做的什么事随意提哈～

Pull request 已创建，详见 #441 ，所需前置配置已在 Pull request 中说明。

> [!WARNING]
>
> 由于缺乏与 **GitHub-hosted Runner** 完全相同的 `windows-latest` 运行时环境，此 YAML 配置文件仅使用 [`nektos/act`](https://github.com/nektos/act) 在 Windows 11 本地进行测试。
>
> 测试过程未有错误，但无法提前保证其绝对正确。仍需待其合并至仓库主分支自动生效后，在下一次发布 Releases 时触发，观察其 Run logs 以确认其行为是否符合预期。

> [!TIP]
>
> 在自动化完全生效前，我会继续人工推送相关更新至 WinGet ❤️
